### PR TITLE
Add filer-s3 implementation for get method

### PR DESCRIPTION
**Pull request by alicoding**

### CODE REVIEW

File: server/util.js [hunk 1]
```diff
@@ -14,6 +14,24 @@ module.exports = {
...
+    for (var i = 0; i < buffer.length; ++i) {
+        view[i] = buffer[i];
+    }
+    return view;
+  },
+
+  toBuffer: function(ab) {
+    var buffer = new Buffer(ab.byteLength);
+    var view = new Uint8Array(ab);
+    for (var i = 0; i < buffer.length; ++i) {
```
Comment (humphd) [inline server/util.js]: Cache the length of `buffer` before you start the loop: `for(var i=0; len=buffer.length; i<len; i++) {`

File: server/util.js [hunk 2]
```diff
@@ -14,6 +14,24 @@ module.exports = {
     var err = new Error( msg );
     err.status = code;
     return err;
+  },
+
+  toArrayBuffer: function(buffer) {
+    var ab = new ArrayBuffer(buffer.length);
+    var view = new Uint8Array(ab);
+    for (var i = 0; i < buffer.length; ++i) {
```
Comment (humphd) [inline server/util.js]: Cache the length of `buffer` before you start the loop: `for(var i=0; len=buffer.length; i<len; i++) {`

**Comment by humphd** [comment]
You've mixed a few things here.  The `get, clear, put, del` code shouldn't be routes.  The only external thing you can do with a path via HTTP is request it like so:

```
https://drive.webmaker.org/path/some/path/to/a/file.txt
```

Everything after `/path/*` is the path into their fs.  You would get it by taking the username (from their cookie/session) and prepending it to the path:  With this, we'd then use filer to get the file, like so:

``` javascript
var username = req.user.username; // or whatever the username is in the cookie
var path = req.path; // or whatever '/some/path/to/a/file.txt' is stored in
var fs = new Filer.FileSystem({provider: new S3Provider({keyPrefix: username}));
fs.readFile(path, function(err, binaryData) {
  if(err) // deal with error...

  // Otherwise set the right mime-time, process the binaryData if necessary, and send on res...
});
```

**Comment by alicoding** [comment]
@humphd I think this patch will really be ready to review is once we have `Filer` in node complete and I have to change it accordingly.

**Comment by humphd** [comment]
That's fine.

File: server/routes.js [hunk 3]
```diff
@@ -19,63 +24,68 @@ module.exports = function( knoxClient ) {
...
+                console.log(error);
+              }
+              res.end();
+            });
+          });
+        });
+      });
+    },
+    get: function(req, res) {
+      var pathToFile = req.path.replace('/get/','');
```
Comment (alicoding) [inline server/routes.js]: @humphd so I've been trying to figure out what would be the right way to do this. I'm trying to strip the first and second `/` in the `path` so I can get the rest of the path to pass in the function call.

There might be a better way to do this, but this is the best way I think we can do here?

Comment (humphd) [inline server/routes.js]: You can use `*` like I do in hubble, see https://github.com/humphd/node-hubble/blob/master/server.js#L19 and https://github.com/humphd/node-hubble/blob/master/routes/index.js#L94

**Comment by alicoding** [comment]
All fixed up for the comments.

**Pull request closed without merging**